wwwapp/settings.py

```diff
@@ -244,3 +244,7 @@
 # Max amount of points that a participant can get during qualification (relative to configured max_points for a given workshop)
 # This allows to give some people bonus points above 100%
 MAX_POINTS_PERCENT = 200
+
+# To protect against escaping INTERNET directory through symlinks and accessing
+# other files set this to INTERNET mount path on application server
+INTERNET_MNT_PATH = None
```

Comment:
This is to prevent espacing via symlinks from one resource folder to other. The files will be served by the same server as Django app so giving this information to Django won't be a problem. It's not feasible to use nginx for this separation as it's not aware of different resource directories. (only of the directory holding all data)

Reply:
I bet you could configure it to not allow symlinks across resource directories. Or just disable symlinks altogether. Trying to implement such protections in a different part of the app than the one handling the request is usually just asking for trouble. What if there is some quirk where Python parses paths differently from nginx? Assuming your current code for parsing the request URI you can trivially bypass all checks with `/internet/www15/symlink?anything=xd` - python will look for a file called `symlink?anything=xd` and not find it while nginx will strip out the query string and serve `symlink`

Reply:
Oh, I think I get it now - you want to configure nginx to give you the full file path AFTER going though symlinks and then verify permissions. I guess that makes sense and should be reasonably secure, if configured properly.

Reply:
Yes, the idea was for nginx to give me full filepath so I can validate if this file can be accessed by currently loggedin user.

Reply:
Can we make sure that the resources are hosted on the same server as the app?